topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -473,6 +475,33 @@ public void setTopicCount(int topics) {
                 action.execute().onComplete(actionResult -> {
                     LOGGER.debug("{}: Executing handler for action {} on topic {}", logContext, action, lockName);
                     action.result = actionResult;
+
+                    String keytag = namespace + ":" + "KafkaTopic" + "/" + key.asKubeName().toString();
+                    Optional<Meter> metric = metrics.meterRegistry().getMeters()
+                            .stream()
+                            .filter(meter -> meter.getId().getName().equals(METRICS_PREFIX + "resource.state") &&
+                                    meter.getId().getTags().contains(Tag.of("kind", "KafkaTopic")) &&
+                                    meter.getId().getTags().contains(Tag.of("name", key.asKubeName().toString())) &&
```

Comment:
You need to get the topic name from its status. I tried your changes and renaming topic ends up with 
```
strimzi_resource_state{kind="KafkaTopic",name="my-topic-2",reason="none",resource_namespace="myproject",} 1.0
strimzi_resource_state{kind="KafkaTopic",name="my-topic-2",reason="Kafka topics cannot be renamed, but KafkaTopic's spec.topicName has changed.",resource_namespace="myproject",} 0.0
```